test/unit/math/rev/functor/gradient_test.cpp

```diff
@@ -35,6 +35,19 @@ TEST(RevFunctor, gradient) {
   EXPECT_FLOAT_EQ(x(0) * x(0) + 3 * 2 * x(1), grad_fx(1));
 }
 
+TEST(RevFunctor, gradient_array) {
+  fun1 f;
+  Matrix<double, Dynamic, 1> x(2);
+  x << 5, 7;
+  double fx;
+  double grad_fx[2];
+  stan::math::gradient(f, x, fx, std::begin(grad_fx), std::end(grad_fx));
```

Comment:
Would prefer not using plain array. Nothing wrong with it but we don't do it anywhere else so if you can change this in other places that would be nice

```suggestion
  std::vector<double> grad_fx(2, 0);
  stan::math::gradient(f, x, fx, std::begin(grad_fx), std::end(grad_fx));
```